src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,349 @@
+calls we currently make into one convenient callable package.
+"""
+
+from pathlib import Path
+from typing import Optional
+
+import pulumi_kubernetes as kubernetes
+from pulumi import ComponentResource, Output, ResourceOptions, debug
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+
+from bridge.lib.magic_numbers import (
+    DEFAULT_NGINX_PORT,
+    DEFAULT_UWSGI_PORT,
+    MAXIMUM_K8S_NAME_LENGTH,
+)
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+
+
+def truncate_k8s_metanames(name: str) -> str:
+    """
+    Sanitize the names we use for k8s objects
+    """
+    truncated_name = name[:MAXIMUM_K8S_NAME_LENGTH].rstrip("-_.")
+    debug(f"moof: {truncated_name=}")
+    return truncated_name
+
+
+class OLApplicationK8sConfiguration(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    project_root: Path
+    application_config: dict[str, str]
+    application_name: str
+    application_namespace: str
+    application_lb_service_name: str
+    application_lb_service_port_name: str
+    k8s_global_labels: dict[str, str]
+    db_creds_secret_name: str
+    redis_creds_secret_name: str
+    static_secrets_name: str
+    application_security_group_id: Output[str]
+    application_security_group_name: Output[str]
+    application_docker_tag: str | None
+    vault_k8s_resource_auth_name: str
+    import_nginx_config: bool
+    resource_requests: dict[str, str] = Field(
+        default={"cpu": "250m", "memory": "300Mi"}
+    )
+    resource_limits: dict[str, str] = Field(
+        default={"cpu": "500m", "memory": "600Mi"},
+    )
+    init_migrations: bool = Field(default=True)
+    init_collectstatic: bool = Field(default=True)
+
+    @field_validator("application_security_group_id")
+    def validate_sec_group_id(application_security_group_id: Output[str]):  # noqa: N805
+        return Output.from_input(application_security_group_id)
+
+    @field_validator("application_security_group_name")
+    def validate_sec_group_name(application_security_group_name: Output[str]):  # noqa: N805
+        return Output.from_input(application_security_group_name)
+
+
+stack_info = parse_stack()
+env_name = f"{stack_info.env_prefix}-{stack_info.env_suffix}"
+
+
+class OLApplicationK8s(ComponentResource):
+    """
+    Main K8s component resource class
+    """
+
+    def __init__(
+        self,
+        ol_app_k8s_config: OLApplicationK8sConfiguration,
+        opts: Optional[ResourceOptions] = None,
+    ):
+        """
+        It's .. the constructor. Shaddap Ruff :)
+        """
+        super().__init__(
+            "ol:infrastructure:components:services:OLApplicationK8s",
+            ol_app_k8s_config.application_namespace,
+            None,
+            opts=opts,
+        )
+        resource_options = ResourceOptions(parent=self).merge(opts)
+        self.ol_app_k8s_config: OLApplicationK8sConfiguration = ol_app_k8s_config
```

Comment:
I don't think we need to assign this to a `self` attribute. There's no real need to access the configs passed in, and it adds extra noise in the access throughout the component (lots of unnecessary `self.`)